### PR TITLE
Boot routers asynchronously.

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

**Comment by ryanpetrello** [comment]
Feel free to look over this and leave comments.  Please don't merge yet, I want to test a bit more strenuously in devstack before we merge this :).

File: akanda/rug/state.py [hunk 1]
```diff
@@ -167,7 +181,7 @@ def transition(self, action, vm, worker_context):
 
 
 class ReadStats(State):
-    def execute(self, action, vm, worker_context, bandwidth_callback):
+    def execute(self, action, vm, worker_context, queue, bandwidth_callback):
```
Comment (dhellmann) [inline akanda/rug/state.py:184]: I think bandwidth_callback is the only special case for arguments now. I wonder if we could just put that in the worker_context.

File: akanda/rug/test/unit/test_state.py [hunk 2]
```diff
@@ -261,9 +288,8 @@ def test_execute(self):
         callback = mock.Mock()
 
         self.assertEqual(
-            self.state.execute(event.READ, self.vm, self.ctx, callback),
-            event.POLL
-        )
+            self.state.execute(event.READ, self.vm, self.ctx, deque(),
+                               callback), event.POLL)
```
Comment (dhellmann) [inline akanda/rug/test/unit/test_state.py]: this indentation seems a little weird, is it pep8-induced?

File: akanda/rug/vm_manager.py [hunk 3]
```diff
@@ -43,6 +46,18 @@ def update_state(self, worker_context, silent=False):
...
         else:
             self.state = DOWN
+            if self.last_boot:
+                seconds_since_boot = (
+                    datetime.utcnow() - self.last_boot
+                ).seconds
+                if seconds_since_boot < cfg.CONF.boot_timeout:
+                    self.state = BOOTING
+                else:
+                    # If the VM was created more than `boot_timeout` seconds
```
Comment (dhellmann) [inline akanda/rug/vm_manager.py:56]: Should we also reset last_boot to None here?

File: akanda/rug/state.py (hunk 1 above)
Comment (ryanpetrello) [inline akanda/rug/state.py:184]: Agreed.  Mind if I do this in a separate commit?  It's going to involve some serious changes to tests.

File: akanda/rug/vm_manager.py (hunk 3 above)
Comment (ryanpetrello) [inline akanda/rug/vm_manager.py:56]: Agreed.

File: akanda/rug/test/unit/test_state.py (hunk 2 above)
Comment (ryanpetrello) [inline akanda/rug/test/unit/test_state.py]: Weird.

File: akanda/rug/state.py (hunk 1 above)
Comment (dhellmann) [inline akanda/rug/state.py:184]: Yeah, that makes sense, I just noticed it and wanted to mention it before I forgot.

File: akanda/rug/test/unit/test_state.py (hunk 2 above)
Comment (dhellmann) [inline akanda/rug/test/unit/test_state.py]: To be clear, I meant the "event.POLL" on line 292.
